lib/debugger/widgets/worldInspect.lua

```diff
@@ -3,74 +3,94 @@ local formatTable = formatTableModule.formatTable
 
 return function(plasma)
 	return plasma.widget(function(debugger, objectStack)
-		local custom = debugger._customWidgets
 		local style = plasma.useStyle()
 
 		local world = debugger.debugWorld
 
 		local cache, setCache = plasma.useState()
+		-- TODO #97 Implement sorting by descending as well.
+		local ascendingOrder, _ = plasma.useState(false)
+		local skipIntersections, setSkipIntersections = plasma.useState(true)
 		local debugComponent, setDebugComponent = plasma.useState()
 
 		local closed = plasma
 			.window({
 				title = "World inspect",
 				closable = true,
 			}, function()
-				local skipIntersections
-
-				plasma.row(function()
-					plasma.heading("Size")
-					plasma.label(world:size())
-
-					plasma.space(30)
-					skipIntersections = plasma.checkbox("Hide intersecting components"):checked()
-
-					if plasma.button("view raw"):clicked() then
-						table.clear(objectStack)
-						objectStack[1] = {
-							value = world,
-							key = "Raw World",
-						}
-					end
-				end)
-
 				if not cache or os.clock() - cache.createdTime > debugger.componentRefreshFrequency then
 					cache = {
 						createdTime = os.clock(),
 						uniqueComponents = {},
+						emptyEntities = 0,
 					}
 
 					setCache(cache)
 
 					for _, entityData in world do
-						for component in entityData do
-							cache.uniqueComponents[component] = (cache.uniqueComponents[component] or 0) + 1
+						if next(entityData) == nil then
+							cache.emptyEntities += 1
+						else
+							for component in entityData do
+								cache.uniqueComponents[component] = (cache.uniqueComponents[component] or 0) + 1
+							end
 						end
 					end
 				end
 
+				plasma.row(function()
+					plasma.heading("Size")
+					plasma.label(
+						`{world:size()} {if cache.emptyEntities > 0 then `({cache.emptyEntities} empty)` else ""}`
+					)
+				end)
+
+				plasma.row({ padding = 15 }, function()
+					if plasma.checkbox("Show intersections", { checked = not skipIntersections }):clicked() then
+						setSkipIntersections(not skipIntersections)
+					end
+
+					if plasma.button("View Raw"):clicked() then
```

Comment:
I think the `View Raw` button should be moved into the same row as the `Size` label.